test/regressions/.mocharc.js

```diff
@@ -1,7 +1,7 @@
-module.exports = {
+export default {
   extension: ['js', 'ts', 'tsx'],
   recursive: true,
   slow: 500,
   timeout: (process.env.CIRCLECI === 'true' ? 5 : 2) * 1000, // Circle CI has low-performance CPUs.
-  require: [require.resolve('../utils/setupBabel')],
+  require: 'tsx',
```

Comment:
👍  Can't do this in core as it has a ton of JSX written in js files

Reply:
The mocharc only applies to server-side files inside `test/regressions` though (basically just `index.test.ts`), which runs in a different context than client-side code (what's imported via `index.tsx`). What comes from the actual codebase is bundled via vite, which is setup to handle JS as JSX. X also has a ton of JSX in JS files, mostly docs code.